src/write.c

```diff
@@ -2400,129 +2358,218 @@ static avifResult avifEncoderWriteMinimizedImageBox(avifEncoder * encoder, avifR
-            (image->matrixCoefficients == (isMonochrome ? AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED : AVIF_MATRIX_COEFFICIENTS_BT601))) {
-            AVIF_CHECKRES(avifRWStreamWriteBits(s, AVIF_MINI_COLOR_TYPE_SRGB, 2)); // bit(2) color_type;
-        } else {
-            const avifMiniColorType colorType = ((image->colorPrimaries >> 5 == 0) && (image->transferCharacteristics >> 5 == 0) &&
-                                                 (image->matrixCoefficients >> 5 == 0))
-                                                    ? AVIF_MINI_COLOR_TYPE_NCLX_5BIT
-                                                    : AVIF_MINI_COLOR_TYPE_NCLX_8BIT;
-            const uint32_t numBitsPerComponent = (colorType == AVIF_MINI_COLOR_TYPE_NCLX_5BIT) ? 5 : 8;
-            AVIF_CHECKRES(avifRWStreamWriteBits(s, colorType, 2));                               // bit(2) color_type;
-            AVIF_CHECKRES(avifRWStreamWriteBits(s, image->colorPrimaries, numBitsPerComponent)); // bit(5/8) color_primaries;
-            AVIF_CHECKRES(avifRWStreamWriteBits(s, image->transferCharacteristics, numBitsPerComponent)); // bit(5/8) transfer_characteristics;
-            if (isMonochrome) {
-                AVIF_CHECKERR(image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED, AVIF_RESULT_INVALID_ARGUMENT);
-            } else {
-                AVIF_CHECKRES(avifRWStreamWriteBits(s, image->matrixCoefficients, numBitsPerComponent)); // bit(5/8) matrix_coefficients;
-            }
-        }
-    }
-
-    AVIF_CHECKRES(avifRWStreamWriteBits(s, 0, 1)); // bit(1) has_explicit_codec_types;
-                                                   // bit(32) infe_type;
-                                                   // bit(32) codec_config_type;
-    AVIF_CHECKRES(avifRWStreamWriteVarInt(s, 4));  // varint main_item_codec_config_size;
-    AVIF_ASSERT_OR_RETURN(colorData->size >= 1);
-    AVIF_CHECKRES(avifRWStreamWriteVarInt(s, (uint32_t)colorData->size - 1)); // varint main_item_data_size_minus_one;
-
-    AVIF_CHECKRES(avifRWStreamWriteBits(s, alphaItem ? 1 : 0, 1)); // bit(1) has_alpha;
-    if (alphaItem) {
-        AVIF_CHECKRES(avifRWStreamWriteBits(s, image->alphaPremultiplied, 1)); // bit(1) alpha_is_premultiplied;
-        AVIF_CHECKRES(avifRWStreamWriteVarInt(s, (uint32_t)alphaData->size));  // varint alpha_item_data_size;
-        AVIF_ASSERT_OR_RETURN(alphaData->size != 0);
-        AVIF_CHECKRES(avifRWStreamWriteVarInt(s, 4)); // varint alpha_item_codec_config_size;
-    }
+    const avifBool fullRange = image->yuvRange == AVIF_RANGE_FULL;
+    const uint32_t pixelFormat = image->depth == 8    ? AVIF_METAV1_PIXEL_FORMAT_UINT8
+                                 : image->depth == 10 ? AVIF_METAV1_PIXEL_FORMAT_UINT10
+                                                      : AVIF_METAV1_PIXEL_FORMAT_UINT12;
+
+    // AV1 does not explicitly signal horizontal chroma siting (could be CSP_RESERVED though).
```

Comment:
This comment seems wrong. If the AV1 `chroma_sample_position` is `CSP_VERTICAL` or `CSP_COLOCATED`, then chroma is NOT horizontally centered.